src/spaceObjects/playerSpaceship.cpp

```diff
@@ -2015,7 +2025,18 @@ void PlayerSpaceship::commandSetScienceLink(P<ScanProbe> probe)
 {
     sf::Packet packet;
     packet << CMD_SET_SCIENCE_LINK;
-    packet << probe->getMultiplayerId();
+
+    // Pass the probe's multiplayer ID if the probe isn't nullptr.
+    if (probe)
+    {
+        packet << probe->getMultiplayerId();
+    }
+    // Otherwise, pass -1 to unlink it, same as commandClearScienceLink().
+    else
+    {
+        LOG(WARNING) << "Attempted to unlink a ScanProbe from Science, but the ScanProbe wasn't valid.";
+        packet << -1;
```

Comment:
Best cast the `-1` to the proper type here, else it's unknown if it adds a 32 or 64 bit number (as integers are allowed to be different sizes)